### PR TITLE
new disable scopes support and basic github actions ci

### PR DESCRIPTION
**Pull request by cyrush**

### CODE REVIEW

File: uberenv.py [hunk 1]
```diff
@@ -719,11 +720,25 @@ def disable_spack_config_scopes(self,spack_dir):
...
+        #
+        # For newer versions of spack, we can use the SPACK_DISABLE_LOCAL_CONFIG
+        # env var plumbing. We patch it to True to make a permanent change.
+        #
+        # Note: This path does not disable the 'site' config, but disabling 'user' config
+        # is our primary goal.
+        #
+        spack_disable_env_stmt = 'disable_local_config = "SPACK_DISABLE_LOCAL_CONFIG" in os.environ'
+        spack_disable_env_stmt_perm = "disable_local_config = True"
+        if cfg_script.count(spack_disable_env_stmt) > 0:
```
Comment (white238) [inline uberenv.py:732]: 👍

**Comment by cyrush** [comment]
I added very basic ci testing as well. We can expand as needed.

**Comment by cyrush** [comment]
resolves #74

**Review by white238 — APPROVED** [review]

**Comment by white238** [comment]
After this gets merged into `master`, we can force that the checks need to pass before being able to merge PRs.

File: .ci/test-project/uberenv_config.json [hunk 2]
```diff
@@ -0,0 +1,10 @@
+{
+"package_name" : "magictestlib",
+"package_version" : "1.0.0",
+"package_source_dir" : ".",
+"spack_build_mode" : "uberenv-pkg",
+"spack_configs_path": "spack_configs",
+"spack_packages_path": "packages",
+"spack_url": "https://github.com/spack/spack",
+"spack_branch": "develop"
```
Comment (kennyweiss) [inline .ci/test-project/uberenv_config.json:9]: Do we want to tie this to `spack@develop` or  lock it down to a specific commit/tag?

File: .github/workflows/linux.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                                curl \
+                                wget \
+                                tar \
+                                unzip \
+                                build-essential
+    - uses: actions/checkout@v2
+    - name: Run Uberenv
+      run: |
+          cd  .ci/test-project
+          python ../../uberenv.py  --project-json=uberenv_config.json
```
Comment (kennyweiss) [inline .github/workflows/linux.yml:32]: Does this fail if uberenv fails? or does it need some logic to fail the CI?

File: .ci/test-project/packages/uberenv-magictestlib/package.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+##############################################################################
```
Comment (kennyweiss) [inline .ci/test-project/packages/uberenv-magictestlib/package.py:1]: Is this one for compatibility with the older workflows?

File: .ci/test-project/packages/magictestlib/package.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    """
+    if vtype is None:
+        if value == "ON" or value == "OFF":
+            vtype = "BOOL"
+        else:
+            vtype = "PATH"
+    return 'set({0} "{1}" CACHE {2} "")\n\n'.format(name, value, vtype)
+
+
+class Magictestlib(Package):
```
Comment (kennyweiss) [inline .ci/test-project/packages/magictestlib/package.py:30]: Should this extend from `CachedCMakePackage` ? 
Or were you going for simplicity?

**Review by kennyweiss — APPROVED** [review]
Thanks @cyrush

File: .ci/test-project/packages/magictestlib/package.py (hunk 5 above)
Comment (cyrush) [reply .ci/test-project/packages/magictestlib/package.py:30]: I don't want to build cmake. Just want to test the basic mechanics.

File: .ci/test-project/packages/uberenv-magictestlib/package.py (hunk 4 above)
Comment (cyrush) [reply .ci/test-project/packages/uberenv-magictestlib/package.py:1]: this test is based on how i use uberenv across projects, but we can also add things that test the dev build but I am not the expert on that way of using.

File: .github/workflows/linux.yml (hunk 3 above)
Comment (cyrush) [reply .github/workflows/linux.yml:32]: the ci fails uberenv fails (or if the cat command fails)

File: .ci/test-project/uberenv_config.json (hunk 2 above)
Comment (cyrush) [reply .ci/test-project/uberenv_config.json:9]: I say we pin *if* we see problems.  The hope is that this is so basic that spack changes shouldn't be able to undermine it.  That might not be the case when we add things like envs, so we can reconsider later.

**Pull request merged**